### PR TITLE
Add support for geometric object plugins, many cleanups

### PR DESCRIPTION
**Pull request by dictoon**
Depends on https://github.com/appleseedhq/appleseed/pull/2626.

### CODE REVIEW

File: src/_beginmaxheaders.h [hunk 1]
```diff
@@ -25,3 +25,9 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 //
+
+// Build options header.
+#include "foundation/core/buildoptions.h"
```
Comment (usakhelo) [inline src/_beginmaxheaders.h]: I like these two headers included here but this header still needs to be included separately in many places.

**Comment by usakhelo** [comment]
I like how cleaner it become, thanks @dictoon . It compiles and runs fine, checked on few of my test scenes.

File: src/_beginmaxheaders.h (hunk 1 above)
Comment (dictoon) [reply src/_beginmaxheaders.h]: As far as I can tell it's included everywhere where it must be, can you point a place where that isn't the case?

Comment (usakhelo) [reply src/_beginmaxheaders.h]: No @dictoon, it's all fine. It's just that _begin header little later includes it again. But that's not big deal. It was just a thought, sorry.

Comment (dictoon) [reply src/_beginmaxheaders.h]: No problem, thanks for the review!

**Pull request merged**